\documentclass[]{article}
\usepackage{lmodern}
\usepackage{amssymb,amsmath}
\usepackage{ifxetex,ifluatex}
\usepackage{fixltx2e} % provides \textsubscript
\ifnum 0\ifxetex 1\fi\ifluatex 1\fi=0 % if pdftex
  \usepackage[T1]{fontenc}
  \usepackage[utf8]{inputenc}
\else % if luatex or xelatex
  \ifxetex
    \usepackage{mathspec}
  \else
    \usepackage{fontspec}
  \fi
  \defaultfontfeatures{Ligatures=TeX,Scale=MatchLowercase}
\fi
% use upquote if available, for straight quotes in verbatim environments
\IfFileExists{upquote.sty}{\usepackage{upquote}}{}
% use microtype if available
\IfFileExists{microtype.sty}{%
\usepackage{microtype}
\UseMicrotypeSet[protrusion]{basicmath} % disable protrusion for tt fonts
}{}
\usepackage[margin=1in]{geometry}
\usepackage{hyperref}
\hypersetup{unicode=true,
            pdftitle={exam.R},
            pdfauthor={franz},
            pdfborder={0 0 0},
            breaklinks=true}
\urlstyle{same}  % don't use monospace font for urls
\usepackage{color}
\usepackage{fancyvrb}
\newcommand{\VerbBar}{|}
\newcommand{\VERB}{\Verb[commandchars=\\\{\}]}
\DefineVerbatimEnvironment{Highlighting}{Verbatim}{commandchars=\\\{\}}
% Add ',fontsize=\small' for more characters per line
\usepackage{framed}
\definecolor{shadecolor}{RGB}{248,248,248}
\newenvironment{Shaded}{\begin{snugshade}}{\end{snugshade}}
\newcommand{\KeywordTok}[1]{\textcolor[rgb]{0.13,0.29,0.53}{\textbf{#1}}}
\newcommand{\DataTypeTok}[1]{\textcolor[rgb]{0.13,0.29,0.53}{#1}}
\newcommand{\DecValTok}[1]{\textcolor[rgb]{0.00,0.00,0.81}{#1}}
\newcommand{\BaseNTok}[1]{\textcolor[rgb]{0.00,0.00,0.81}{#1}}
\newcommand{\FloatTok}[1]{\textcolor[rgb]{0.00,0.00,0.81}{#1}}
\newcommand{\ConstantTok}[1]{\textcolor[rgb]{0.00,0.00,0.00}{#1}}
\newcommand{\CharTok}[1]{\textcolor[rgb]{0.31,0.60,0.02}{#1}}
\newcommand{\SpecialCharTok}[1]{\textcolor[rgb]{0.00,0.00,0.00}{#1}}
\newcommand{\StringTok}[1]{\textcolor[rgb]{0.31,0.60,0.02}{#1}}
\newcommand{\VerbatimStringTok}[1]{\textcolor[rgb]{0.31,0.60,0.02}{#1}}
\newcommand{\SpecialStringTok}[1]{\textcolor[rgb]{0.31,0.60,0.02}{#1}}
\newcommand{\ImportTok}[1]{#1}
\newcommand{\CommentTok}[1]{\textcolor[rgb]{0.56,0.35,0.01}{\textit{#1}}}
\newcommand{\DocumentationTok}[1]{\textcolor[rgb]{0.56,0.35,0.01}{\textbf{\textit{#1}}}}
\newcommand{\AnnotationTok}[1]{\textcolor[rgb]{0.56,0.35,0.01}{\textbf{\textit{#1}}}}
\newcommand{\CommentVarTok}[1]{\textcolor[rgb]{0.56,0.35,0.01}{\textbf{\textit{#1}}}}
\newcommand{\OtherTok}[1]{\textcolor[rgb]{0.56,0.35,0.01}{#1}}
\newcommand{\FunctionTok}[1]{\textcolor[rgb]{0.00,0.00,0.00}{#1}}
\newcommand{\VariableTok}[1]{\textcolor[rgb]{0.00,0.00,0.00}{#1}}
\newcommand{\ControlFlowTok}[1]{\textcolor[rgb]{0.13,0.29,0.53}{\textbf{#1}}}
\newcommand{\OperatorTok}[1]{\textcolor[rgb]{0.81,0.36,0.00}{\textbf{#1}}}
\newcommand{\BuiltInTok}[1]{#1}
\newcommand{\ExtensionTok}[1]{#1}
\newcommand{\PreprocessorTok}[1]{\textcolor[rgb]{0.56,0.35,0.01}{\textit{#1}}}
\newcommand{\AttributeTok}[1]{\textcolor[rgb]{0.77,0.63,0.00}{#1}}
\newcommand{\RegionMarkerTok}[1]{#1}
\newcommand{\InformationTok}[1]{\textcolor[rgb]{0.56,0.35,0.01}{\textbf{\textit{#1}}}}
\newcommand{\WarningTok}[1]{\textcolor[rgb]{0.56,0.35,0.01}{\textbf{\textit{#1}}}}
\newcommand{\AlertTok}[1]{\textcolor[rgb]{0.94,0.16,0.16}{#1}}
\newcommand{\ErrorTok}[1]{\textcolor[rgb]{0.64,0.00,0.00}{\textbf{#1}}}
\newcommand{\NormalTok}[1]{#1}
\usepackage{graphicx,grffile}
\makeatletter
\def\maxwidth{\ifdim\Gin@nat@width>\linewidth\linewidth\else\Gin@nat@width\fi}
\def\maxheight{\ifdim\Gin@nat@height>\textheight\textheight\else\Gin@nat@height\fi}
\makeatother
% Scale images if necessary, so that they will not overflow the page
% margins by default, and it is still possible to overwrite the defaults
% using explicit options in \includegraphics[width, height, ...]{}
\setkeys{Gin}{width=\maxwidth,height=\maxheight,keepaspectratio}
\IfFileExists{parskip.sty}{%
\usepackage{parskip}
}{% else
\setlength{\parindent}{0pt}
\setlength{\parskip}{6pt plus 2pt minus 1pt}
}
\setlength{\emergencystretch}{3em}  % prevent overfull lines
\providecommand{\tightlist}{%
  \setlength{\itemsep}{0pt}\setlength{\parskip}{0pt}}
\setcounter{secnumdepth}{0}
% Redefines (sub)paragraphs to behave more like sections
\ifx\paragraph\undefined\else
\let\oldparagraph\paragraph
\renewcommand{\paragraph}[1]{\oldparagraph{#1}\mbox{}}
\fi
\ifx\subparagraph\undefined\else
\let\oldsubparagraph\subparagraph
\renewcommand{\subparagraph}[1]{\oldsubparagraph{#1}\mbox{}}
\fi

%%% Use protect on footnotes to avoid problems with footnotes in titles
\let\rmarkdownfootnote\footnote%
\def\footnote{\protect\rmarkdownfootnote}

%%% Change title format to be more compact
\usepackage{titling}

% Create subtitle command for use in maketitle
\newcommand{\subtitle}[1]{
  \posttitle{
    \begin{center}\large#1\end{center}
    }
}

\setlength{\droptitle}{-2em}

  \title{exam.R}
    \pretitle{\vspace{\droptitle}\centering\huge}
  \posttitle{\par}
    \author{franz}
    \preauthor{\centering\large\emph}
  \postauthor{\par}
      \predate{\centering\large\emph}
  \postdate{\par}
    \date{Thu Oct 4 11:12:08 2018}


\begin{document}
\maketitle

\begin{Shaded}
\begin{Highlighting}[]
\CommentTok{#!/usr/bin/Rscript}
\CommentTok{#Cours: Analyses génomiques}
\CommentTok{#Prof: Karine Audouze }
\CommentTok{#Université Paris Diderot ~ Master Biologie Informatique M2}

\CommentTok{#Author: AKE Franz-Arnold ~ Pastelle}
\CommentTok{#Date: 28/09/2018}

\CommentTok{#Set current working directory}
\KeywordTok{setwd}\NormalTok{(}\DataTypeTok{dir=} \StringTok{"~/Master/Courses_M2BI/analyses_genomiq/src/"}\NormalTok{)}

\CommentTok{# import libraries}
\KeywordTok{source}\NormalTok{(}\StringTok{"https://bioconductor.org/biocLite.R"}\NormalTok{)}
\end{Highlighting}
\end{Shaded}

\begin{verbatim}
## Bioconductor version 3.6 (BiocInstaller 1.28.0), ?biocLite for help
\end{verbatim}

\begin{verbatim}
## A new version of Bioconductor is available after installing the most
##   recent version of R; see http://bioconductor.org/install
\end{verbatim}

\begin{Shaded}
\begin{Highlighting}[]
\KeywordTok{require}\NormalTok{(oligo)}
\end{Highlighting}
\end{Shaded}

\begin{verbatim}
## Loading required package: oligo
\end{verbatim}

\begin{verbatim}
## Loading required package: BiocGenerics
\end{verbatim}

\begin{verbatim}
## Loading required package: parallel
\end{verbatim}

\begin{verbatim}
## 
## Attaching package: 'BiocGenerics'
\end{verbatim}

\begin{verbatim}
## The following objects are masked from 'package:parallel':
## 
##     clusterApply, clusterApplyLB, clusterCall, clusterEvalQ,
##     clusterExport, clusterMap, parApply, parCapply, parLapply,
##     parLapplyLB, parRapply, parSapply, parSapplyLB
\end{verbatim}

\begin{verbatim}
## The following objects are masked from 'package:stats':
## 
##     IQR, mad, sd, var, xtabs
\end{verbatim}

\begin{verbatim}
## The following objects are masked from 'package:base':
## 
##     anyDuplicated, append, as.data.frame, cbind, colMeans,
##     colnames, colSums, do.call, duplicated, eval, evalq, Filter,
##     Find, get, grep, grepl, intersect, is.unsorted, lapply,
##     lengths, Map, mapply, match, mget, order, paste, pmax,
##     pmax.int, pmin, pmin.int, Position, rank, rbind, Reduce,
##     rowMeans, rownames, rowSums, sapply, setdiff, sort, table,
##     tapply, union, unique, unsplit, which, which.max, which.min
\end{verbatim}

\begin{verbatim}
## Loading required package: oligoClasses
\end{verbatim}

\begin{verbatim}
## Welcome to oligoClasses version 1.40.0
\end{verbatim}

\begin{verbatim}
## Loading required package: Biobase
\end{verbatim}

\begin{verbatim}
## Welcome to Bioconductor
## 
##     Vignettes contain introductory material; view with
##     'browseVignettes()'. To cite Bioconductor, see
##     'citation("Biobase")', and for packages 'citation("pkgname")'.
\end{verbatim}

\begin{verbatim}
## Loading required package: Biostrings
\end{verbatim}

\begin{verbatim}
## Loading required package: S4Vectors
\end{verbatim}

\begin{verbatim}
## Loading required package: stats4
\end{verbatim}

\begin{verbatim}
## 
## Attaching package: 'S4Vectors'
\end{verbatim}

\begin{verbatim}
## The following object is masked from 'package:base':
## 
##     expand.grid
\end{verbatim}

\begin{verbatim}
## Loading required package: IRanges
\end{verbatim}

\begin{verbatim}
## Loading required package: XVector
\end{verbatim}

\begin{verbatim}
## 
## Attaching package: 'Biostrings'
\end{verbatim}

\begin{verbatim}
## The following object is masked from 'package:base':
## 
##     strsplit
\end{verbatim}

\begin{verbatim}
## ===========================================================================
\end{verbatim}

\begin{verbatim}
## Welcome to oligo version 1.42.0
\end{verbatim}

\begin{verbatim}
## ===========================================================================
\end{verbatim}

\begin{Shaded}
\begin{Highlighting}[]
\KeywordTok{require}\NormalTok{(annotate)}
\end{Highlighting}
\end{Shaded}

\begin{verbatim}
## Loading required package: annotate
\end{verbatim}

\begin{verbatim}
## Loading required package: AnnotationDbi
\end{verbatim}

\begin{verbatim}
## Loading required package: XML
\end{verbatim}

\begin{Shaded}
\begin{Highlighting}[]
\KeywordTok{require}\NormalTok{(genefilter)}
\end{Highlighting}
\end{Shaded}

\begin{verbatim}
## Loading required package: genefilter
\end{verbatim}

\begin{Shaded}
\begin{Highlighting}[]
\CommentTok{#library(affy)}
\KeywordTok{library}\NormalTok{(pd.mogene.}\FloatTok{2.0}\NormalTok{.st)}
\end{Highlighting}
\end{Shaded}

\begin{verbatim}
## Loading required package: RSQLite
\end{verbatim}

\begin{verbatim}
## Loading required package: DBI
\end{verbatim}

\begin{Shaded}
\begin{Highlighting}[]
\KeywordTok{library}\NormalTok{(mogene20sttranscriptcluster.db)}
\end{Highlighting}
\end{Shaded}

\begin{verbatim}
## Loading required package: org.Mm.eg.db
\end{verbatim}

\begin{verbatim}
## 
\end{verbatim}

\begin{verbatim}
## 
\end{verbatim}

\begin{Shaded}
\begin{Highlighting}[]
\KeywordTok{library}\NormalTok{(hgu133a.db)}
\end{Highlighting}
\end{Shaded}

\begin{verbatim}
## Loading required package: org.Hs.eg.db
\end{verbatim}

\begin{verbatim}
## 
\end{verbatim}

\begin{verbatim}
## 
\end{verbatim}

\begin{Shaded}
\begin{Highlighting}[]
\KeywordTok{library}\NormalTok{(FactoMineR)}
\KeywordTok{library}\NormalTok{(factoextra)}
\end{Highlighting}
\end{Shaded}

\begin{verbatim}
## Loading required package: ggplot2
\end{verbatim}

\begin{verbatim}
## Welcome! Related Books: `Practical Guide To Cluster Analysis in R` at https://goo.gl/13EFCZ
\end{verbatim}

\begin{Shaded}
\begin{Highlighting}[]
\KeywordTok{library}\NormalTok{(rmarkdown)}
\KeywordTok{library}\NormalTok{(ensembldb)}
\end{Highlighting}
\end{Shaded}

\begin{verbatim}
## Loading required package: GenomicRanges
\end{verbatim}

\begin{verbatim}
## Loading required package: GenomeInfoDb
\end{verbatim}

\begin{verbatim}
## Loading required package: GenomicFeatures
\end{verbatim}

\begin{verbatim}
## Loading required package: AnnotationFilter
\end{verbatim}

\begin{verbatim}
## 
## Attaching package: 'ensembldb'
\end{verbatim}

\begin{verbatim}
## The following object is masked from 'package:stats':
## 
##     filter
\end{verbatim}

\begin{Shaded}
\begin{Highlighting}[]
\KeywordTok{library}\NormalTok{(}\StringTok{"RDAVIDWebService"}\NormalTok{)}
\end{Highlighting}
\end{Shaded}

\begin{verbatim}
## Loading required package: graph
\end{verbatim}

\begin{verbatim}
## 
## Attaching package: 'graph'
\end{verbatim}

\begin{verbatim}
## The following object is masked from 'package:XML':
## 
##     addNode
\end{verbatim}

\begin{verbatim}
## The following object is masked from 'package:Biostrings':
## 
##     complement
\end{verbatim}

\begin{verbatim}
## Loading required package: GOstats
\end{verbatim}

\begin{verbatim}
## Loading required package: Category
\end{verbatim}

\begin{verbatim}
## Loading required package: Matrix
\end{verbatim}

\begin{verbatim}
## 
## Attaching package: 'Matrix'
\end{verbatim}

\begin{verbatim}
## The following object is masked from 'package:S4Vectors':
## 
##     expand
\end{verbatim}

\begin{verbatim}
## 
\end{verbatim}

\begin{verbatim}
## 
## Attaching package: 'GOstats'
\end{verbatim}

\begin{verbatim}
## The following object is masked from 'package:AnnotationDbi':
## 
##     makeGOGraph
\end{verbatim}

\begin{verbatim}
## 
## Attaching package: 'RDAVIDWebService'
\end{verbatim}

\begin{verbatim}
## The following object is masked from 'package:ensembldb':
## 
##     genes
\end{verbatim}

\begin{verbatim}
## The following objects are masked from 'package:GenomicFeatures':
## 
##     genes, species
\end{verbatim}

\begin{verbatim}
## The following object is masked from 'package:GenomeInfoDb':
## 
##     species
\end{verbatim}

\begin{verbatim}
## The following object is masked from 'package:AnnotationDbi':
## 
##     species
\end{verbatim}

\begin{verbatim}
## The following object is masked from 'package:Biostrings':
## 
##     type
\end{verbatim}

\begin{verbatim}
## The following object is masked from 'package:IRanges':
## 
##     members
\end{verbatim}

\begin{verbatim}
## The following objects are masked from 'package:BiocGenerics':
## 
##     counts, species
\end{verbatim}

\begin{Shaded}
\begin{Highlighting}[]
\KeywordTok{library}\NormalTok{(gplots)}
\end{Highlighting}
\end{Shaded}

\begin{verbatim}
## 
## Attaching package: 'gplots'
\end{verbatim}

\begin{verbatim}
## The following object is masked from 'package:IRanges':
## 
##     space
\end{verbatim}

\begin{verbatim}
## The following object is masked from 'package:S4Vectors':
## 
##     space
\end{verbatim}

\begin{verbatim}
## The following object is masked from 'package:stats':
## 
##     lowess
\end{verbatim}

\begin{Shaded}
\begin{Highlighting}[]
\CommentTok{#Reading Data CeL (needed mogene library)}
\NormalTok{data.Cel =}\StringTok{ }\KeywordTok{read.celfiles}\NormalTok{(}\DataTypeTok{filenames =} \KeywordTok{sort}\NormalTok{(}\KeywordTok{list.celfiles}\NormalTok{(}\StringTok{"../data/CEL-20180928/Archive/"}\NormalTok{, }\DataTypeTok{full =} \OtherTok{TRUE}\NormalTok{)))}
\end{Highlighting}
\end{Shaded}

\begin{verbatim}
## Platform design info loaded.
\end{verbatim}

\begin{verbatim}
## Reading in : ../data/CEL-20180928/Archive//1-A_(MoGene-2_0-st).CEL
## Reading in : ../data/CEL-20180928/Archive//10-J_(MoGene-2_0-st).CEL
## Reading in : ../data/CEL-20180928/Archive//2-B_(MoGene-2_0-st).CEL
## Reading in : ../data/CEL-20180928/Archive//3-C_(MoGene-2_0-st).CEL
## Reading in : ../data/CEL-20180928/Archive//4-D_(MoGene-2_0-st).CEL
## Reading in : ../data/CEL-20180928/Archive//5-E_(MoGene-2_0-st).CEL
## Reading in : ../data/CEL-20180928/Archive//6-F_(MoGene-2_0-st).CEL
## Reading in : ../data/CEL-20180928/Archive//7-G_(MoGene-2_0-st).CEL
## Reading in : ../data/CEL-20180928/Archive//8-H_(MoGene-2_0-st).CEL
## Reading in : ../data/CEL-20180928/Archive//9-I_(MoGene-2_0-st).CEL
\end{verbatim}

\begin{Shaded}
\begin{Highlighting}[]
\CommentTok{#normalisation des données}
\NormalTok{ei =}\StringTok{ }\KeywordTok{exprs}\NormalTok{(}\KeywordTok{rma}\NormalTok{(data.Cel))}
\end{Highlighting}
\end{Shaded}

\begin{verbatim}
## Background correcting
## Normalizing
## Calculating Expression
\end{verbatim}

\begin{Shaded}
\begin{Highlighting}[]
\CommentTok{#Annotate Genes names}
\NormalTok{data.Cel.ID <-}\StringTok{ }\KeywordTok{rownames}\NormalTok{(ei)}
\NormalTok{GeneNames <-}\StringTok{ }\KeywordTok{unlist}\NormalTok{(}\KeywordTok{contents}\NormalTok{(mogene20sttranscriptclusterGENENAME)[data.Cel.ID])}
\NormalTok{GeneSymb <-}\StringTok{ }\KeywordTok{unlist}\NormalTok{(}\KeywordTok{contents}\NormalTok{(mogene20sttranscriptclusterSYMBOL)[data.Cel.ID])}
\NormalTok{GeneEnsembl <-}\StringTok{ }\KeywordTok{unlist}\NormalTok{(}\KeywordTok{contents}\NormalTok{(mogene20sttranscriptclusterENSEMBL)[data.Cel.ID])}

\CommentTok{#Visualisation}
\KeywordTok{par}\NormalTok{(}\DataTypeTok{mfrow =} \KeywordTok{c}\NormalTok{(}\DecValTok{1}\NormalTok{,}\DecValTok{2}\NormalTok{))}
\KeywordTok{plot}\NormalTok{(}\KeywordTok{density}\NormalTok{(ei[,}\DecValTok{1}\NormalTok{], }\DataTypeTok{na.rm =} \OtherTok{TRUE}\NormalTok{), }\DataTypeTok{xlim =} \KeywordTok{c}\NormalTok{(}\DecValTok{0}\NormalTok{,}\DecValTok{16}\NormalTok{), }\DataTypeTok{ylim =} \KeywordTok{c}\NormalTok{(}\DecValTok{0}\NormalTok{,}\FloatTok{0.3}\NormalTok{), main}
\NormalTok{     =}\StringTok{ "Normalized Data"}\NormalTok{, }\DataTypeTok{xlab =} \StringTok{"Intensity"}\NormalTok{)}
\ControlFlowTok{for}\NormalTok{(i }\ControlFlowTok{in} \DecValTok{2}\OperatorTok{:}\KeywordTok{ncol}\NormalTok{( ei ) ) \{}
  \KeywordTok{points}\NormalTok{( }\KeywordTok{density}\NormalTok{( ei[,i], }\DataTypeTok{na.rm =} \OtherTok{TRUE}\NormalTok{ ), }\DataTypeTok{type =} \StringTok{"l"}\NormalTok{, }\DataTypeTok{col =} \KeywordTok{rainbow}\NormalTok{(}\DecValTok{20}\NormalTok{)[i] )}
\NormalTok{\}}
\KeywordTok{plot}\NormalTok{(}\KeywordTok{density}\NormalTok{(ei[,}\DecValTok{1}\OperatorTok{:}\DecValTok{9}\NormalTok{], }\DataTypeTok{na.rm =} \OtherTok{TRUE}\NormalTok{), }\DataTypeTok{xlim =} \KeywordTok{c}\NormalTok{(}\DecValTok{0}\NormalTok{,}\DecValTok{16}\NormalTok{), }\DataTypeTok{ylim =} \KeywordTok{c}\NormalTok{(}\DecValTok{0}\NormalTok{,}\FloatTok{0.3}\NormalTok{), main}
\NormalTok{     =}\StringTok{ "Normalized Data without sample I"}\NormalTok{, }\DataTypeTok{xlab =} \StringTok{"Intensity"}\NormalTok{)}
\ControlFlowTok{for}\NormalTok{(i }\ControlFlowTok{in} \DecValTok{2}\OperatorTok{:}\KeywordTok{ncol}\NormalTok{( ei )}\OperatorTok{-}\DecValTok{1}\NormalTok{) \{}
  \KeywordTok{points}\NormalTok{( }\KeywordTok{density}\NormalTok{( ei[,i], }\DataTypeTok{na.rm =} \OtherTok{TRUE}\NormalTok{ ), }\DataTypeTok{type =} \StringTok{"l"}\NormalTok{, }\DataTypeTok{col =} \KeywordTok{rainbow}\NormalTok{(}\DecValTok{20}\NormalTok{)[i] )}
\NormalTok{\}}
\end{Highlighting}
\end{Shaded}

\includegraphics{exam_files/figure-latex/unnamed-chunk-1-1.pdf}

\begin{Shaded}
\begin{Highlighting}[]
\KeywordTok{dev.off}\NormalTok{()}
\end{Highlighting}
\end{Shaded}

\begin{verbatim}
## null device 
##           1
\end{verbatim}

\begin{Shaded}
\begin{Highlighting}[]
\KeywordTok{cat}\NormalTok{(}\StringTok{"}
\StringTok{    Q1:}
\StringTok{    We can observe (plot 1) that all the sample are normalized and centred except sample I which have a bad fit}
\StringTok{    with others sample (plot 2). It must be an outlier.}
\StringTok{"}\NormalTok{)}
\end{Highlighting}
\end{Shaded}

\begin{verbatim}
## 
##     Q1:
##     We can observe (plot 1) that all the sample are normalized and centred except sample I which have a bad fit
##     with others sample (plot 2). It must be an outlier.
\end{verbatim}

\begin{Shaded}
\begin{Highlighting}[]
\KeywordTok{cat}\NormalTok{(}\StringTok{"Q2 : Combien   de  gènes   uniques sont    présents? ~ Pourquoi    certains    sont    ils répliqués ? "}\NormalTok{)}
\end{Highlighting}
\end{Shaded}

\begin{verbatim}
## Q2 : Combien de  gènes   uniques sont    présents? ~ Pourquoi    certains    sont    ils répliqués ?
\end{verbatim}

\begin{Shaded}
\begin{Highlighting}[]
\NormalTok{nb.unique.genes =}\StringTok{  }\KeywordTok{length}\NormalTok{(GeneNames[}\OperatorTok{!}\NormalTok{(}\KeywordTok{duplicated}\NormalTok{(GeneNames)}\OperatorTok{|}\KeywordTok{duplicated}\NormalTok{(GeneNames, }\DataTypeTok{fromLast=}\OtherTok{TRUE}\NormalTok{))])}
\KeywordTok{cat}\NormalTok{(}\KeywordTok{paste}\NormalTok{(}\StringTok{"we have"}\NormalTok{,nb.unique.genes,}\StringTok{"unique genes"}\NormalTok{))}
\end{Highlighting}
\end{Shaded}

\begin{verbatim}
## we have 23516 unique genes
\end{verbatim}

\begin{Shaded}
\begin{Highlighting}[]
\KeywordTok{cat}\NormalTok{(}\StringTok{"}
\StringTok{    we have multiples same genes because the probes match probably}
\StringTok{    multiples same transcript of the same gene}
\StringTok{"}\NormalTok{)}
\end{Highlighting}
\end{Shaded}

\begin{verbatim}
## 
##     we have multiples same genes because the probes match probably
##     multiples same transcript of the same gene
\end{verbatim}

\begin{Shaded}
\begin{Highlighting}[]
\KeywordTok{cat}\NormalTok{(}\StringTok{"Q3:  Réaliser  une PCA     Y   a   t’il    un  échantillon ‘outlier’ ? si  oui lequel. Dans    ce  cas ne  pas }
\StringTok{ l’utiliser pour    la  suite   des analyses."}\NormalTok{)}
\end{Highlighting}
\end{Shaded}

\begin{verbatim}
## Q3:  Réaliser    une PCA     Y   a   t’il    un  échantillon ‘outlier’ ? si  oui lequel. Dans    ce  cas ne  pas 
##  l’utiliser  pour    la  suite   des analyses.
\end{verbatim}

\begin{Shaded}
\begin{Highlighting}[]
\NormalTok{design <-}\StringTok{ }\KeywordTok{c}\NormalTok{(}\KeywordTok{rep}\NormalTok{(}\KeywordTok{c}\NormalTok{(}\StringTok{"heal"}\NormalTok{,}\StringTok{"sick"}\NormalTok{), }\DataTypeTok{each=}\DecValTok{1}\NormalTok{, }\DecValTok{3}\NormalTok{), }\KeywordTok{rep}\NormalTok{(}\KeywordTok{c}\NormalTok{(}\StringTok{"heal"}\NormalTok{),}\DecValTok{2}\NormalTok{), }\KeywordTok{rep}\NormalTok{(}\KeywordTok{c}\NormalTok{(}\StringTok{"sick"}\NormalTok{),}\DecValTok{2}\NormalTok{))}
\NormalTok{design <-}\StringTok{ }\KeywordTok{data.frame}\NormalTok{(design)}
\NormalTok{data.Cel.design <-}\StringTok{ }\KeywordTok{cbind}\NormalTok{(design, }\KeywordTok{t}\NormalTok{(ei))  }\CommentTok{#concatenate design with ei}
\KeywordTok{rownames}\NormalTok{(data.Cel.design) =}\StringTok{ }\KeywordTok{c}\NormalTok{(}\StringTok{"A"}\NormalTok{, }\StringTok{"J"}\NormalTok{, }\StringTok{"B"}\NormalTok{, }\StringTok{"C"}\NormalTok{, }\StringTok{"D"}\NormalTok{, }\StringTok{"E"}\NormalTok{, }\StringTok{"F"}\NormalTok{, }\StringTok{"G"}\NormalTok{, }\StringTok{"H"}\NormalTok{, }\StringTok{"I"}\NormalTok{)  }\CommentTok{#rename samples rows}
\KeywordTok{colnames}\NormalTok{(ei) =}\StringTok{ }\KeywordTok{rownames}\NormalTok{(data.Cel.design)}
\NormalTok{data.Cel.PCA =}\StringTok{ }\KeywordTok{PCA}\NormalTok{(data.Cel.design, }\DataTypeTok{ncp=}\DecValTok{5}\NormalTok{, }\DataTypeTok{quali.sup =} \DecValTok{1}\NormalTok{, }\DataTypeTok{graph =}\NormalTok{ F)  }\CommentTok{#compute PCA}

\CommentTok{#Visualisation PCA ~ With the Outlier I}
\CommentTok{#Percentage of cumulative Variance}
\CommentTok{#barplot(data.Cel.PCA$eig[,1], main = "Eboulis des Eigen Values of ACP")}
\KeywordTok{fviz_screeplot}\NormalTok{(data.Cel.PCA, }\DataTypeTok{addlabels =} \OtherTok{TRUE}\NormalTok{, }\DataTypeTok{ylim =} \KeywordTok{c}\NormalTok{(}\DecValTok{0}\NormalTok{, }\DecValTok{50}\NormalTok{),}
               \DataTypeTok{main =} \StringTok{"Evolution de la variance expliquée ~ Dimensions"}\NormalTok{)}
\CommentTok{# plot.PCA(data.Cel.PCA, axes = c(1,4), choix = "ind", habillage = 1)}
\KeywordTok{fviz_pca_ind}\NormalTok{(data.Cel.PCA, }\DataTypeTok{habillage =}\NormalTok{ data.Cel.design[,}\DecValTok{1}\NormalTok{], }\DataTypeTok{addEllipses =}\NormalTok{ F)}

\CommentTok{#Echantillon_Outlier}
\KeywordTok{cat}\NormalTok{(}\StringTok{"Oui nous avons un echantillon outlier, en l'occurence l'echantillon I}
\StringTok{      Elimination de l'echantillon outlier..."}\NormalTok{)}
\end{Highlighting}
\end{Shaded}

\begin{verbatim}
## Oui nous avons un echantillon outlier, en l'occurence l'echantillon I
##       Elimination de l'echantillon outlier...
\end{verbatim}

\begin{Shaded}
\begin{Highlighting}[]
\NormalTok{ei =}\StringTok{ }\NormalTok{ei[,}\DecValTok{1}\OperatorTok{:}\DecValTok{9}\NormalTok{]}
\NormalTok{design <-}\StringTok{ }\KeywordTok{c}\NormalTok{(}\KeywordTok{rep}\NormalTok{(}\KeywordTok{c}\NormalTok{(}\StringTok{"heal"}\NormalTok{,}\StringTok{"sick"}\NormalTok{), }\DataTypeTok{each=}\DecValTok{1}\NormalTok{, }\DecValTok{3}\NormalTok{), }\KeywordTok{rep}\NormalTok{(}\KeywordTok{c}\NormalTok{(}\StringTok{"heal"}\NormalTok{),}\DecValTok{2}\NormalTok{), }\KeywordTok{rep}\NormalTok{(}\KeywordTok{c}\NormalTok{(}\StringTok{"sick"}\NormalTok{),}\DecValTok{1}\NormalTok{))}
\NormalTok{design <-}\StringTok{ }\KeywordTok{data.frame}\NormalTok{(design)}
\NormalTok{data.Cel.design <-}\StringTok{ }\KeywordTok{cbind}\NormalTok{(design, }\KeywordTok{t}\NormalTok{(ei))  }\CommentTok{#concatenate design with ei}
\KeywordTok{rownames}\NormalTok{(data.Cel.design) =}\StringTok{ }\KeywordTok{c}\NormalTok{(}\StringTok{"A"}\NormalTok{, }\StringTok{"J"}\NormalTok{, }\StringTok{"B"}\NormalTok{, }\StringTok{"C"}\NormalTok{, }\StringTok{"D"}\NormalTok{, }\StringTok{"E"}\NormalTok{, }\StringTok{"F"}\NormalTok{, }\StringTok{"G"}\NormalTok{, }\StringTok{"H"}\NormalTok{)  }\CommentTok{#rename samples rows}


\KeywordTok{cat}\NormalTok{(}\StringTok{"Q4: Réaliser   un  volcano plot.   Sélectionner    un  seuil.  Pourquoi    avez-vous   choisi ce   seuil ?"}\NormalTok{)}
\end{Highlighting}
\end{Shaded}

\begin{verbatim}
## Q4: Réaliser un  volcano plot.   Sélectionner    un  seuil.  Pourquoi    avez-vous   choisi ce   seuil ?
\end{verbatim}

\begin{Shaded}
\begin{Highlighting}[]
\CommentTok{#T-tests}
\NormalTok{t.pval =}\StringTok{ }\KeywordTok{rowttests}\NormalTok{(ei,design}\OperatorTok{$}\NormalTok{design)}\OperatorTok{$}\NormalTok{p.value}
\KeywordTok{names}\NormalTok{(t.pval) =}\StringTok{ }\KeywordTok{rownames}\NormalTok{(ei)}

\CommentTok{#Fold_change}
\NormalTok{fc_cel =}\StringTok{ }\KeywordTok{rowMeans}\NormalTok{(ei[, design}\OperatorTok{$}\NormalTok{design }\OperatorTok{==}\StringTok{ "heal"}\NormalTok{]) }\OperatorTok{-}\StringTok{ }\KeywordTok{rowMeans}\NormalTok{(ei[, design}\OperatorTok{$}\NormalTok{design }\OperatorTok{==}\StringTok{ "sick"}\NormalTok{])}

\CommentTok{#Random}
\NormalTok{random.labels =}\StringTok{ }\KeywordTok{sample}\NormalTok{(design}\OperatorTok{$}\NormalTok{design)}
\NormalTok{perm_t.pval =}\StringTok{ }\KeywordTok{rowttests}\NormalTok{(ei, random.labels)}\OperatorTok{$}\NormalTok{p.value}
\NormalTok{perm_fc_cel =}\StringTok{ }\KeywordTok{rowMeans}\NormalTok{(ei[, random.labels }\OperatorTok{==}\StringTok{ "heal"}\NormalTok{]) }\OperatorTok{-}\StringTok{ }\KeywordTok{rowMeans}\NormalTok{(ei[, random.labels }\OperatorTok{==}\StringTok{ "sick"}\NormalTok{])}

\CommentTok{#Visualisation}
\KeywordTok{plot}\NormalTok{(fc_cel, t.pval, }\DataTypeTok{main =} \StringTok{"Volcano Plot"}\NormalTok{, }\DataTypeTok{log =} \StringTok{"y"}\NormalTok{, }\DataTypeTok{xlab =} \StringTok{"M (log2 fold change)"}\NormalTok{, }
     \DataTypeTok{ylab =} \StringTok{"p-value"}\NormalTok{, }\DataTypeTok{pch =} \DecValTok{20}\NormalTok{, }\DataTypeTok{col =} \StringTok{"blue"}\NormalTok{, }\DataTypeTok{xlim =} \KeywordTok{c}\NormalTok{(}\OperatorTok{-}\FloatTok{2.5}\NormalTok{,}\FloatTok{2.5}\NormalTok{))}
\KeywordTok{points}\NormalTok{(perm_fc_cel, perm_t.pval, }\DataTypeTok{type =} \StringTok{"p"}\NormalTok{, }\DataTypeTok{pch =} \DecValTok{20}\NormalTok{, }\DataTypeTok{col =} \StringTok{"red"}\NormalTok{)}
\KeywordTok{legend}\NormalTok{(}\StringTok{"topleft"}\NormalTok{, }\DataTypeTok{col=}\KeywordTok{c}\NormalTok{(}\StringTok{"red"}\NormalTok{,}\StringTok{"blue"}\NormalTok{), }\DataTypeTok{legend=}\KeywordTok{c}\NormalTok{(}\StringTok{"perm"}\NormalTok{, }\StringTok{"real"}\NormalTok{), }\DataTypeTok{pch=}\DecValTok{5}\NormalTok{, }\DataTypeTok{bg=}\StringTok{"white"}\NormalTok{, }\DataTypeTok{cex =} \FloatTok{0.5}\NormalTok{)}

\KeywordTok{abline}\NormalTok{(}\DataTypeTok{v=}\OperatorTok{-}\DecValTok{2}\NormalTok{, }\DataTypeTok{col=}\StringTok{"black"}\NormalTok{, }\DataTypeTok{lty=}\DecValTok{4}\NormalTok{, }\DataTypeTok{lwd=}\FloatTok{2.0}\NormalTok{)}
\KeywordTok{abline}\NormalTok{(}\DataTypeTok{v=}\DecValTok{2}\NormalTok{, }\DataTypeTok{col=}\StringTok{"black"}\NormalTok{, }\DataTypeTok{lty=}\DecValTok{4}\NormalTok{, }\DataTypeTok{lwd=}\FloatTok{2.0}\NormalTok{)}
\KeywordTok{abline}\NormalTok{(}\DataTypeTok{h=}\FloatTok{0.008}\NormalTok{, }\DataTypeTok{col=}\StringTok{"black"}\NormalTok{, }\DataTypeTok{lty=}\DecValTok{4}\NormalTok{, }\DataTypeTok{lwd=}\FloatTok{2.0}\NormalTok{) }
\KeywordTok{cat}\NormalTok{(}\StringTok{"threshold défini à 0.008"}\NormalTok{)}
\end{Highlighting}
\end{Shaded}

\begin{verbatim}
## threshold défini à 0.008
\end{verbatim}

\begin{Shaded}
\begin{Highlighting}[]
\CommentTok{#why ?}

\CommentTok{#Selection of significatives genes}
\NormalTok{ei_top =}\StringTok{ }\NormalTok{ei[}\KeywordTok{names}\NormalTok{(t.pval[t.pval}\OperatorTok{<}\FloatTok{0.001}\NormalTok{]),] }\CommentTok{#we have 37 genes}
\KeywordTok{rownames}\NormalTok{(ei_top) =}\StringTok{ }\NormalTok{GeneSymb[}\KeywordTok{rownames}\NormalTok{(ei_top)]}
\KeywordTok{cat}\NormalTok{(}\KeywordTok{paste}\NormalTok{(}\StringTok{"we have"}\NormalTok{, }\KeywordTok{dim}\NormalTok{(ei_top)[}\DecValTok{1}\NormalTok{], }\StringTok{"genes"}\NormalTok{))}
\end{Highlighting}
\end{Shaded}

\begin{verbatim}
## we have 52 genes
\end{verbatim}

\begin{Shaded}
\begin{Highlighting}[]
\CommentTok{#Q6 - Realiser un enrichissement biologique (GSEA) ~ dbase KEGG ~ tool DAVID}



\CommentTok{#Q7 - Compute HeatMap}
\CommentTok{#visualisation}
\NormalTok{scaleyellowred <-}\StringTok{ }\KeywordTok{colorRampPalette}\NormalTok{(}\KeywordTok{c}\NormalTok{(}\StringTok{"white"}\NormalTok{, }\StringTok{"green"}\NormalTok{, }\StringTok{"blue"}\NormalTok{, }\StringTok{"orange"}\NormalTok{, }\StringTok{"red"}\NormalTok{), }\DataTypeTok{space =} \StringTok{"rgb"}\NormalTok{)(}\DecValTok{256}\NormalTok{)}
\KeywordTok{heatmap.2}\NormalTok{(ei_top, }\DataTypeTok{col=}\NormalTok{scaleyellowred, }\DataTypeTok{trace =} \StringTok{"none"}\NormalTok{, }\DataTypeTok{density.info =} \StringTok{"none"}\NormalTok{)}
\end{Highlighting}
\end{Shaded}


\end{document}
